lua/lint/parser.lua

```diff
@@ -22,53 +21,108 @@ function M.from_errorformat(efm, skeleton)
           },
           message = item.text,
         }
-        table.insert(
-          result,
-          vim.tbl_extend('keep', diagnostic, skeleton and skeleton or defaults)
-        )
+        table.insert(result, vim.tbl_extend('keep', diagnostic, skeleton and skeleton or defaults))
       end
     end
     return result
   end
 end
 
+--- Parse a linter's output using a regex pattern
+-- @param pattern The regex pattern
+-- @param groups The groups defined by the pattern: {"lineno", "message", ("colno" | "colbeg", "colend"), ["code"], ["codeDesc"], ["file"], ["severity"]}
+-- @param severity_map An optional table mapping the severity values to their codes
+-- @param defaults An optional table of diagnostic default values
+function M.from_pattern(pattern, groups, severity_map, defaults)
+  local group_handler = {
+    code = function(entries)
+      return entries['code']
+    end,
 
--- Return a parse function that uses a pattern to parse the output.
---
--- The first argument is a lua pattern.
--- The pattern must match 4 groups in order: line number, offset, code and message
---
--- The second argument is a skeleton, used to create each diagnostic.
--- It should contain default values - for example the `source` and `severity`.
---
--- The output of the linter must have 1 entry per line
-function M.from_pattern(pattern, diagnostic_skeleton)
-  return function(output)
-    local result = vim.fn.split(output, "\n")
+    codeDescription = function(entries)
+      return entries['codeDesc']
+    end,
+
+    data = function(_)
+      return nil -- Unsupported
+    end,
+
+    message = function(entries)
+      return entries['message']
+    end,
+
+    range = function(entries)
+      local lineno = tonumber(entries['lineno'])
+      if entries['colbeg'] ~= nil and entries['colend'] ~= nil then
+        local colbeg, colend = tonumber(entries['colbeg']), tonumber(entries['colend'])
+        return {
+          ['start'] = { line = lineno - 1, character = colbeg - 1 },
+          ['end'] = { line = lineno - 1, character = colend },
+        }
+      end
+
+      local colno = entries['colno'] ~= nil and tonumber(entries['colno']) or 1
+      return {
+        ['start'] = { line = lineno - 1, character = colno - 1 },
+        ['end'] = { line = lineno - 1, character = colno },
+      }
+    end,
+
+    relatedInformation = function(_)
```

Comment:
Why do we already need methods for stuff that is not supported?

Reply:
It was just to show how it should be implemented. Actually, I could just return entries[...] and it would be implemented, but as it was not used I didn't bother :).
Should I has support since it's easy to do ?

Reply:
I'd remove it if it is currently not used by any parser. We can extend it once needed.